tests/unittests/test_datasource/test_ec2.py

```diff
@@ -267,30 +383,24 @@ def _setup_ds(self, sys_cfg, platform_data, md, md_version=None):
                         register_mock_metaserver(instance_id_url, None)
         return ds
 
-    def test_network_config_property_returns_version_1_network_data(self):
-        """network_config property returns network version 1 for metadata.
-
-        Only one device is configured even when multiple exist in metadata.
-        """
+    def test_network_config_property_returns_version_2_network_data(self):
+        """network_config property returns network version 2 for metadata"""
         ds = self._setup_ds(
             platform_data=self.valid_platform_data,
             sys_cfg={'datasource': {'Ec2': {'strict_id': True}}},
             md={'md': DEFAULT_METADATA})
-        find_fallback_path = (
-            'cloudinit.sources.DataSourceEc2.net.find_fallback_nic')
+        find_fallback_path = M_PATH_NET + 'find_fallback_nic'
         with mock.patch(find_fallback_path) as m_find_fallback:
             m_find_fallback.return_value = 'eth9'
             ds.get_data()
 
         mac1 = '06:17:04:d7:26:09'  # Defined in DEFAULT_METADATA
-        expected = {'version': 1, 'config': [
-            {'mac_address': '06:17:04:d7:26:09', 'name': 'eth9',
-             'subnets': [{'type': 'dhcp4'}, {'type': 'dhcp6'}],
-             'type': 'physical'}]}
-        patch_path = (
-            'cloudinit.sources.DataSourceEc2.net.get_interfaces_by_mac')
-        get_interface_mac_path = (
-            'cloudinit.sources.DataSourceEc2.net.get_interface_mac')
+        expected = {'version': 2, 'ethernets': {'eth9': {
+            'match': {'macaddress': '06:17:04:d7:26:09'}, 'set-name': 'eth9',
+            'dhcp4': True, 'dhcp4-overrides': {'route-metric': 100},
```

Comment:
In the case where we are only DHCPing on an interface, do we need to apply the route metrics at all?

My (perhaps naive) assumption is that the vast majority of instances will be in this case, and we could avoid changing the generated config for them entirely if we omit the route metrics when they aren't needed.

What do you think?

Reply:
Yes I think this makes sense as us adding a route-metric where someone else may have added secondary nics post-cloud-init setup without route-metrics would actually leave our fallback nic network config being deprioritised. So, I think it better to leave it without the overrides if cloud-init doesn't see other nics.